src/System Application/App/Entity Text/src/EntityTextImpl.Codeunit.al

```diff
@@ -50,19 +56,24 @@ codeunit 2012 "Entity Text Impl."
             Error(CapabilityDisabledErr);
         if not CanSuggest() then
             Error(CannotGenerateErr);
-        BuildPrompts(Facts, Tone, TextFormat, TextEmphasis, SystemPrompt, UserPrompt);
+        BuildPrompts(Facts, Tone, TextFormat, TextEmphasis, SystemPrompt, UserPrompt, SourceScenario);
 
         Session.LogMessage('0000JVG', TelemetryGenerationRequestedTxt, Verbosity::Normal, DataClassification::SystemMetadata, TelemetryScope::ExtensionPublisher, 'Category', TelemetryCategoryLbl);
 
-        Suggestion := GenerateAndReviewCompletion(SystemPrompt, UserPrompt, TextFormat, Facts, CallerModuleInfo);
+        Suggestion := GenerateAndReviewCompletion(SystemPrompt, UserPrompt, TextFormat, Facts, CallerModuleInfo, SourceScenario);
 
         exit(Suggestion);
     end;
 
     procedure InsertSuggestion(SourceTableId: Integer; SourceSystemId: Guid; SourceScenario: Enum "Entity Text Scenario"; SuggestedText: Text)
     var
         EntityText: Record "Entity Text";
+        Handled: Boolean;
     begin
+        OnInsertSuggestion(SourceTableId, SourceSystemId, SourceScenario, SuggestedText, Handled);
```

Comment:
I don't agree with this event. 

Using events will allow any extension to potentially break existing copilots. We should refactor to have a way to control this before we even get here. Such that Marketing Text Suggestions or any copilot that uses `GenerateSuggestion` will always have their intended flow.

Reply:
Event removed in latest iteration